textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -118,7 +126,14 @@ public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList,
      */
     public static Schema createSpanSchema(Schema schema) {
         List<Attribute> dataTupleAttributes = schema.getAttributes();
-        //spanAttributes contains all attributes of dataTupleAttributes and an additional SPAN_LIST_ATTRIBUTE
+        // If SourceOperator is of type Keyword, spanschema is already provided

```

Comment:
Again, if you know you don't need to add span, why do you call `createSpanSchema` in the first place? Is it better to put this logic in dictionaryMatcher?